### PR TITLE
Validating education benefits form against schema

### LINKED ISSUE #3082: Check edu benefits application against json schema
Will need to:

- [ ] Add schema as a submodule to vets-website
- [ ] Add ajv module
- [ ] Write tests that take form data, convert that data to the application format, and validate the result against the schema.

We should figure a good way to get good test coverage for this. Some sort of property-based testing?

### PR DESCRIPTION
**Pull request by jbalboni**
Closes #3082.

This is an attempt at adding schema validation to our tests for edu benefits.

I wanted to try to do more comprehensive testing than we would get from hand creating a bunch of test cases for validation, so I'm using quick_check.js to do property-based testing. I'm using it to generate random (valid) data in the shape of our veteran form data, converting that to an application, then validating the result.

I also wrote some tests to make sure that the generated data matches the shape of our blank veteran form in the front end to hopefully avoid the data in the actual front end application diverging from the generated data.

Any thoughts on this approach are appreciated!

### CODE REVIEW

**Review by markolson — APPROVED** [review]
I like the idea of validating the test structure against the `createVeteran` structure right off the bat. Changing the test structure I'm able to get failures, so 👍 from me

**Comment by jbalboni** [comment]
@alexose Can you take a look when you get a chance?

**Comment by jbalboni** [comment]
Here are two of my concerns about this:

1. Generating test data is pretty slow (can take 2+ seconds)
2. There is some randomness to the generated data, so theoretically we could have a failure on one machine but not another. Probably very unlikely, though.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "schema"]
```
Comment (alexose) [inline .gitmodules]: Do we need to update the `vets-website` to tell people to install submodules?

**Review by alexose — APPROVED** [review]
I like this strategy.  I'm curious why generating data is slow-- While there are a lot of fields, it seems like it shouldn't be _that_ slow.

**Comment by jbalboni** [comment]
@alexose I think the slowness (and I mean slow in that it takes 1-2 seconds to run 100 generated test cases, so not that bad) is a result of all the steps, i.e. generating the data, parsing a json string, and validating against the schema.

File: .gitmodules (hunk 1 above)
Comment (jbalboni) [inline .gitmodules]: I should check, but I don't think people have to do anything different for things to work. But I should probably add a note about how to update the submodule.

Comment (jbalboni) [inline .gitmodules]: Made the updates.

**Pull request merged**